commcare_export/excel_query.py

```diff
@@ -258,14 +258,47 @@ def compile_source(worksheet):
     api_query = Apply(*api_query_args)
 
     if data_source_jsonpath is None or isinstance(data_source_jsonpath, jsonpath.This) or isinstance(data_source_jsonpath, jsonpath.Root):
-        return api_query, None
+        return data_source, api_query, None
     else:
-        return api_query, Reference(str(data_source_jsonpath))
+        return data_source, api_query, Reference(str(data_source_jsonpath))
+
+# If the source is expected to provide a column, then require that it is
+# already present or can be added without conflicting with an existing
+# column.
+def require_column_in_sheet(sheet_name, data_source, table_name, output_headings,
+                            output_fields, column_enforcer):
+    # Check for conflicting use of column name.
+    extend_fields = True
+
+    required_column = column_enforcer.column_to_require(data_source)
+    if required_column is None:
+        extend_fields = False
+    else:
+        for i in range(len(output_headings)):
+            if output_headings[i].value == required_column.name.v:
+                if isinstance(output_fields[i], Reference) and \
+                   output_fields[i].ref == required_column.source:
```

Comment:
nit: recommended indentation it two-levels for wrapped continuation
```suggestion
                if isinstance(output_fields[i], Reference) and \
                        output_fields[i].ref == required_column.source:
```